ecommers/lib/core/provider_models/cart_provider.dart

```diff
@@ -0,0 +1,67 @@
+import 'package:ecommers/core/models/index.dart';
+import 'package:ecommers/core/services/dependency_service.dart';
+import 'package:flutter/widgets.dart';
+
+class CartProvider extends ChangeNotifier {
+  List<OrderModel> _orders;
+
+  List<OrderModel> get orders => _orders;
+  int get orderCount => _orders.length;
+
+  Future removeOrEdit(OrderModel order) async {
+    final editOrder = _getOrderById(order.id);
+
+    if (editOrder == null) {
+      return;
+    }
+
+    if (editOrder.count > 1) {
+      editOrder.count--;
+      await cartRepository.editOrder(order);
+    } else {
+      _orders.remove(editOrder);
+      await cartRepository.removeOrder(order);
+    }
+    notifyListeners();
+  }
+
+  Future addOrEdit(OrderModel order) async{
+    final editOrder = _getOrderById(order.id);
+
+    if (editOrder == null) {
+      _orders.add(order);
+      await cartRepository.addOrder(order);
+    } else {
+      editOrder.count++;
+      await cartRepository.editOrder(order);
+    }
+    notifyListeners();
+  }
+
+  void resetCart() {
+    _orders = <OrderModel>[];
+    cartRepository.dropOrders();
+    notifyListeners();
+  }
+
+  double calculateTotalCost([double salePercent = 0]) {
```

Comment:
salePercent - do we use this argument somewhere?

Reply:
checkoutPage have sale